src/hamster/session.py

```diff
@@ -35,34 +37,36 @@ def __init__(self):
-            "/org/gnome/SessionManager")
-        self.__session_manager_iface = dbus.Interface(session_manager,
-            dbus_interface="org.gnome.SessionManager")
-        self.__client_id = self.__session_manager_iface.RegisterClient("", "")
-
-        # Get SessionManager.ClientPrivate interface
-        session_client = session_bus.get_object("org.gnome.SessionManager",
-            self.__client_id)
-        self.__session_client_private_iface = dbus.Interface(session_client,
-            dbus_interface="org.gnome.SessionManager.ClientPrivate")
-
-        # Connect to the needed signals
-        session_bus.add_signal_receiver(self.__query_end_session_handler,
-            signal_name = "QueryEndSession",
-            dbus_interface = "org.gnome.SessionManager.ClientPrivate",
-            bus_name = "org.gnome.SessionManager")
-
-        session_bus.add_signal_receiver(self.__end_session_handler,
-            signal_name = "EndSession",
-            dbus_interface = "org.gnome.SessionManager.ClientPrivate",
-            bus_name = "org.gnome.SessionManager")
-
-        session_bus.add_signal_receiver(self.__stop_handler,
-            signal_name = "Stop",
-            dbus_interface = "org.gnome.SessionManager.ClientPrivate",
-            bus_name = "org.gnome.SessionManager")
-
+        try:
+            session_manager = session_bus.get_object("org.gnome.SessionManager",
+                "/org/gnome/SessionManager")
+            self.__session_manager_iface = dbus.Interface(session_manager,
+                dbus_interface="org.gnome.SessionManager")
+            self.__client_id = self.__session_manager_iface.RegisterClient("", "")
+
+            # Get SessionManager.ClientPrivate interface
+            session_client = session_bus.get_object("org.gnome.SessionManager",
+                self.__client_id)
+            self.__session_client_private_iface = dbus.Interface(session_client,
+                dbus_interface="org.gnome.SessionManager.ClientPrivate")
+
+            # Connect to the needed signals
+            session_bus.add_signal_receiver(self.__query_end_session_handler,
+                signal_name = "QueryEndSession",
+                dbus_interface = "org.gnome.SessionManager.ClientPrivate",
+                bus_name = "org.gnome.SessionManager")
+
+            session_bus.add_signal_receiver(self.__end_session_handler,
+                signal_name = "EndSession",
+                dbus_interface = "org.gnome.SessionManager.ClientPrivate",
+                bus_name = "org.gnome.SessionManager")
+
+            session_bus.add_signal_receiver(self.__stop_handler,
+                signal_name = "Stop",
+                dbus_interface = "org.gnome.SessionManager.ClientPrivate",
+                bus_name = "org.gnome.SessionManager")
+        except dbus.exceptions.DBusException:
+            logger.warning("Unable to connect to GNOME session manager, stop trcking on logout won't work")
```

Comment:
Please either make this logger.info() or just replace this line with `pass` so command line users don't see this message on every single command (because warning is the default logging level). Also trcking -> tracking.

Reply:
Ok, I have changed it but I think this message will appear only when your start the hamster daemon so probably not on each command.

Reply:
Ah right, I forgot that I was using a cover script that restart the daemon before each command. Thanks anyway.